### PR TITLE
print lightning log to local file

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
https://github.com/pingcap/tidb-lightning/issues/307
Now lightning print logs directly to console if the user doesn't set log path. The stack info will confuse the users. For example:
```log
[2020/05/09 22:44:18.681 +08:00] [ERROR] [main.go:76] ["tidb lightning encountered error"] [error="Error 1062: Duplicate entry '120' for key 'gen_idx'"] [errorVerbose="Error 1062: Duplicate entry '120' for key 'gen_idx'\ng
ithub.com/pingcap/errors.AddStack\n\t/home/pingcap/goPath/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/errors.go:174\ngithub.com/pingcap/errors.Trace\n\t/home/pingcap/goPath/pkg/mod/github.com/pi
ngcap/errors@v0.11.5-0.20190809092503-95897b64e011/juju_adaptor.go:15\ngithub.com/pingcap/tidb-lightning/lightning/restore.(*chunkRestore).restore\n\t/home/pingcap/goPath/src/github.com/pingcap/tidb-lightning/lightning/res
tore/restore.go:1824\ngithub.com/pingcap/tidb-lightning/lightning/restore.(*TableRestore).restoreEngine.func1\n\t/home/pingcap/goPath/src/github.com/pingcap/tidb-lightning/lightning/restore/restore.go:934\nruntime.goexit\n
\t/usr/local/go/src/runtime/asm_amd64.s:1357"]
```

### What is changed and how it works?
Redirect logs to log file. If it's not given, will save log to `./logs/tidb-lightning-timestamp.log`. Only print necessary error message in console.

The testing result is:
```
Verbose debug logs will be written to /home/root/bin/logs/tidb-lightning-2020-05-11-09-52-43.log.

Welcome to lightning
Release Version: v3.0.14-4-g632582e
Git Commit Hash: 632582e34007807bd9f6d3656cb5094574a2c67c
Git Branch: fixLogStack
UTC Build Time: 2020-05-11 01:51:59
Go Version: go version go1.13.4 linux/amd64

loaded task's cfg: {"id":1589161963977751803,"lightning":{"table-concurrency":16,"index-concurrency":16,"region-concurrency":16,"io-concurrency":5,"check-requirements":false},"tidb":{"host":"127.0.0.1","port":4000,"user":"
root","status-port":10080,"pd-addr":"/tmp/tidb","sql-mode":"ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION","tls":"false","security
":{"ca-path":"","cert-path":"","key-path":""},"max-allowed-packet":67108864,"distsql-scan-concurrency":100,"build-stats-concurrency":20,"index-serial-scan-concurrency":20,"checksum-table-concurrency":16},"checkpoint":{"ena
ble":true,"schema":"tidb_lightning_checkpoint","driver":"file","keep-after-success":false},"mydumper":{"read-block-size":65536,"batch-size":107374182400,"batch-import-ratio":0,"no-schema":false,"character-set":"auto","csv":{"separator":",","delimiter":"\"","header":true,"trim-last-separator":false,"not-null":false,"null":"\\N","backslash-escape":tru
e,"max-region-size":0},"case-sensitive":false,"strict-format":false,"max-region-size":268435456},"black-white-list":{"do-tables":null,"do-dbs":null,"ignore-tables":null,"ignore-dbs":["mysql","information_schema","performan
ce_schema","sys"]},"tikv-importer":{"addr":"","backend":"tidb","on-duplicate":"error","max-kv-pairs":32},"post-restore":{"level-1-compact":false,"compact":false,"checksum":true,"analyze":true},"cron":{"switch-mode":"5m0s",
"log-progress":"5m0s"},"routes":null,"security":{"ca-path":"","cert-path":"","key-path":""}}

Error: restore table `all_mode`.`t1` failed: Error 1062: Duplicate entry '120' for key 'gen_idx'
tidb lightning encountered error:  restore table `all_mode`.`t1` failed: Error 1062: Duplicate entry '120' for key 'gen_idx'
```

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Related changes

 - Need to be included in the release note

### CODE REVIEW

**Comment by kennytm** [comment]
Some printf will affect DM who uses Lightning as a library.

File: lightning/common/version.go [hunk 1]
```diff
@@ -52,6 +53,7 @@ func PrintInfo(app string, callback func()) {
 		zap.String("UTC Build Time", BuildTS),
 		zap.String("Go Version", GoVersion),
 	)
+	fmt.Fprintln(os.Stdout, "Welcome to "+app+"\n"+GetRawInfo())
```
Comment (kennytm) [inline lightning/common/version.go:56]: let's move the welcome log to `main()`, or just remove it.

File: lightning/config/global.go [hunk 2]
```diff
@@ -180,6 +180,13 @@ func LoadGlobalConfig(args []string, extraFlags func(*flag.FlagSet)) (*GlobalCon
 	}
 	if *logFilePath != "" {
 		cfg.App.Config.File = *logFilePath
+	} else {
+		configFile, err := log.AdjustLogFilePath()
+		if err != nil {
+			return nil, errors.Annotate(err, "Create debug logs directory failed")
+		}
+		cfg.App.Config.File = configFile
+		fmt.Fprintf(os.Stdout, "Verbose debug logs will be written to %s.\n\n", configFile)
```
Comment (kennytm) [inline lightning/config/global.go:189]: there should still be a way to explicitly use `stderr` as log output.

i suggest using BR's approach of making the default value of the CLI parameter to be the log path.

File: lightning/lightning.go [hunk 3]
```diff
@@ -171,6 +171,7 @@ var taskCfgRecorderKey struct{}
 func (l *Lightning) run(taskCfg *config.Config) (err error) {
 	common.PrintInfo("lightning", func() {
 		log.L().Info("cfg", zap.Stringer("cfg", taskCfg))
+		fmt.Fprintf(os.Stdout, "loaded task's cfg: %s\n\n",  taskCfg)
```
Comment (kennytm) [inline lightning/lightning.go:174]: no point printing this to stdout

File: lightning/restore/restore.go [hunk 4]
```diff
@@ -1246,7 +1247,7 @@ func (rc *RestoreController) cleanCheckpoints(ctx context.Context) error {
 		err = rc.checkpointsDB.RemoveCheckpoint(ctx, "all")
 	}
 	task.End(zap.ErrorLevel, err)
-	return errors.Trace(err)
+	return errors.Trace(errors.Annotate(err, "clean checkpoints"))
```
Comment (kennytm) [inline lightning/restore/restore.go:1250]:
```suggestion
	return errors.Annotate(err, "clean checkpoints")
```

annotate already implies trace.

**Comment by lichunzhu** [comment]
> Some printf will affect DM who uses Lightning as a library.

I move all `Fprintf` to main.go.

File: cmd/tidb-lightning/main.go [hunk 5]
```diff
@@ -73,8 +76,10 @@ func main() {
 	}
 	if err != nil {
 		logger.Error("tidb lightning encountered error", zap.Error(err))
+		fmt.Fprintln(os.Stderr, "tidb lightning encountered error: ", err)
 	} else {
 		logger.Info("tidb lightning exit")
+		fmt.Fprintln(os.Stdout, "tidb lightning exit", err)
```
Comment (kennytm) [inline cmd/tidb-lightning/main.go:82]:
```suggestion
		fmt.Fprintln(os.Stdout, "tidb lightning exit")
```

**Pull request merged**